joomla/joomla-cms
# Mcrypt disabled warning in Two Factor authentication

Pull Request for Issue https://github.com/joomla/joomla-cms/issues/9507.
### Summary of Changes

This solves the php fatal error in https://github.com/joomla/joomla-cms/issues/9507 by adding a warning message when mcrypt php extension is not enabled.
### Testing Instructions

Before applying this PR patch is the issue described in https://github.com/joomla/joomla-cms/issues/9507.

Test if 2FA still works as before.
1. Install Joomla with latest staging
2. Apply patch
3. Disable mcrypt and restart server
4. Test configuring 2FA (see below) - you will get a warning message.
5. Enable mcrypt and restart server
6. Test configuring 2FA (see below)
7. Now logout and try to login with 2FA
8. Code review

Note: Test Frontend (profile) and Backend user edit.
### How to test 2FA
- Enable Google Authenticator two factor authentication plugin: Extensions > Plugins (twofactorauth)
- Create a dummy user and configure it to use with Google Authenticator: Users > Manage > Edit (2FA tab)

Note: you can use a chrome plugin for configuring and using 2FA, for instance https://chrome.google.com/webstore/detail/authenticator/bhghoamapcdpbohphigoooaddinpkbai
#### Observations

Other PR related:
- https://github.com/joomla/joomla-cms/pull/9532
- https://github.com/joomla/joomla-cms/pull/9508
